shop/src/components/Layout.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { PureComponent } from 'react';
+import { Outlet } from "react-router-dom";
+import { BsFillCartPlusFill } from 'react-icons/bs';
+import ShopIcons from './ShopIcons';
+import Quantity from './Quantity';
+import style from './Layout.module.css';
+import { connect } from 'react-redux';
+import * as consts from './actions/actions';
+
+class Layout extends PureComponent {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            displayIcons: false,
+            image: null,
+        }
+    }
+
+    showDisplayIcons = () => {
+        if (isNaN(this.productPrice.value)) {
+            alert('price is number !!!');
+            return;
+        }
+        if (this.productName.value) {
+            this.setState({
+                displayIcons: !this.state.displayIcons,
+            });
+        }
+    };
+
+    setMoreCount = () => {
+        this.props.addCount();
+    };
+
+    setLessCount = () => {
+        if (this.props.count !== 1) {
```

Comment:
this approach is also good!

I usually use "early return", the idea is pretty simple:
```
if (this.props.count === 1) {
  return;
}
```
It means that you don't care about the rest code below in this function